testplan/runners/pools/base.py

```diff
@@ -447,7 +473,9 @@ def _handle_taskresults(self, worker, request, response):
                 "De-assign {} from {}".format(task_result.task, worker)
             )
 
-            if self.should_reschedule(self, task_result):
+            if self.should_reschedule(self, task_result) or self.should_rerun(
```

Comment:
I think the rerun logic could be implemented as an default implementation of should_reschedule. User can override if desired.

I also suggest that we use term 'rerun' in here, we can save 'reschedule' for assigning task from one pool to another.